datafusion/functions-aggregate/src/count.rs

```diff
@@ -467,7 +467,7 @@ impl GroupsAccumulator for CountGroupsAccumulator {
         &mut self,
         values: &[ArrayRef],
         group_indices: &[usize],
-        opt_filter: Option<&BooleanArray>,
+        _opt_filter: Option<&BooleanArray>,
```

Comment:
I recommend we put a comment here explaining why we are ignoring the filter.

In general if your assumption holds, we could actually remove the `opt_filter` argument for all calls to `merge_batch` 🤔

Reply:
That is why I return error for now, we can remove the filter field after couple of months

Reply:
Filed https://github.com/apache/datafusion/issues/13484 to track 👍